rusk-recovery/src/state.rs

```diff
@@ -136,15 +136,16 @@ where
     Ok(state_id)
 }
 
-pub fn exec(build: bool) -> Result<(), Box<dyn Error>> {
+pub fn exec(build: bool, force: bool) -> Result<(), Box<dyn Error>> {
     let theme = Theme::default();
 
     info!("{} Network state", theme.action("Checking"));
     let state_path = rusk_profile::get_rusk_state_dir()?;
     let id_path = rusk_profile::get_rusk_state_id_path()?;
 
-    // if the state already exists in the expected path stop.
-    if state_path.exists() && id_path.exists() {
+    // if we're not forcing a rebuild/download and the state already exists in
+    // the expected path, stop early.
+    if !force && state_path.exists() && id_path.exists() {
```

Comment:
What about `deploy`? Should it affected too?

Also, I believe this changed should come after the one where we integrate the "data" at build time, especially because `deploy` – currently we're "deploying" only if we're building but that is not entirely correct, since `deploy` is always related to the local state instead of the one integrated in `rusk-recovery`.

Reply:
The intended behavior is as follows:

| Outcome       | Download | Download | Build | Build | Stop | Download | Stop | Build |
|---------------|:----------:|:----------:|:-------:|:-------:|:------:|:----------:|:------:|:-------:|
| State Present | No       | No       | No    | No    | Yes  | Yes      | Yes  | Yes   |
| Build Flag    | No       | No       | Yes   | Yes   | No   | No       | Yes  | Yes   |
| Force Flag    | No       | Yes      | No    | Yes   | No   | Yes      | No   | Yes   |

The `deploy` function is actually `build`ing the state, so I don't believe any changes are required here. It is also used independently by other parts of the code, so I don't believe its behavior should be changed. Maybe its name could change, since it is building a new state. I would suggest, however, that this would be the a different issue.

Reply:
Ok, if I understood the header is not exactly an header in the previous table, so it would be something like:

| State Present | Build Flag | Force Flag | Outcome |
| - | - | - | - |
| No           | No         | No         | Download
| No           | No         | Yes        | Download
| No           | Yes        | No         | Build
| No           | Yes        | Yes        | Build
| Yes          | No         | No         | Stop
| Yes          | No         | Yes        | Download
| Yes          | Yes        | No         | Stop
| Yes          | Yes        | Yes        | Build

Is that correct?